src/types/instantsearch.ts

```diff
@@ -13,16 +13,15 @@ export type HelperChangeEvent = {
 
 type HitAttributeHighlightResult = {
   value: string;
-  matchLevel: 'none' | 'partial' | 'full';
+  matchLevel: string;
   matchedWords: string[];
   fullyHighlighted?: boolean;
 };
 
 type HitHighlightResult = {
   [attribute: string]:
     | HitAttributeHighlightResult
-    | HitAttributeHighlightResult[]
-    | HitHighlightResult;
```

Comment:
This type was recursive, therefore was unusable.

Reply:
unusable for what? the real type is recursive, no?

Reply:
Interesting, here is working: http://www.typescriptlang.org/play/?ssl=1&ssc=1&pln=16&pc=30#code/C4TwDgpgBAEglsAgsYAnOAjArsC8DmAFgDZxHABKEAzlscFALxQDeAUFJ1AG4CGxWCAC4o1NHAB2+ANxsAvmzahIsBARJlClGnQbN2nANq8U6bLhFj0UgLpCOXAD6qkpzDjybS5KrXoPOZ3hXcXNPIm8tX11DGwCoILUvTW0-YFk5WTYAMywJAGNgOAB7CShs4uKAClQdehFg9UjU3QBKVnj5RQrqlnLKkT6+AWEoAHIesag5OVagA.

Let me revisit this.

Reply:
reverted.